syft/pkg/cataloger/golang/parse_go_bin.go

```diff
@@ -1,26 +1,53 @@
+//nolint
 package golang
 
 import (
-	"bufio"
+	"bytes"
+	"debug/elf"
+	"debug/macho"
+	"debug/pe"
+	"errors"
 	"fmt"
 	"io"
+	"runtime/debug"
 	"strings"
 
+	"github.com/anchore/syft/internal/log"
 	"github.com/anchore/syft/syft/pkg"
+	"github.com/anchore/syft/syft/pkg/cataloger/golang/internal/xcoff"
 	"github.com/anchore/syft/syft/source"
 )
 
-const (
-	packageIdentifier = "dep"
-	replaceIdentifier = "=>"
+const GOARCH = "GOARCH"
+
+var (
+	// errUnrecognizedFormat is returned when a given executable file doesn't
+	// appear to be in a known format, or it breaks the rules of that format,
+	// or when there are I/O errors reading the file.
+	errUnrecognizedFormat = errors.New("unrecognized file format")
```

Comment:
this isn't a very helpful error message; we should convey more context here (what was attempted to be parsed? what was expected? is there another error that we should be wrapping here?)

Reply:
right, I am now wrapping the returned error with more context and only using this error when no other info is available.